### DISCUSSION TITLE
Network unsupported problem occurred while switching connection links

### DISCUSSION THREAD

**Discussion by quankuiyu** [wevm/wagmi#1554] [Q&A]
network does not support ENS (operation="getAvatar", network="matic", code=UNSUPPORTED_OPERATION, version=providers/5.7.2)

**Comment by tmm** [comment]
What Chain(s) are you using? If it's not `mainnet` or `goerli`, they don't support ENS and you either need to deploy the ENS contracts or target another Chain.

**Reply by 0xxmemo** [reply]
```
Unhandled Runtime Error
Error: unsupported network (argument="network", value={"chainId":137,"name":"polygon"}, code=INVALID_ARGUMENT, version=providers/5.7.2)
```
```
// Wagmi
import { configureChains, createClient, WagmiConfig } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'
import { infuraProvider } from 'wagmi/providers/infura'
import { publicProvider } from 'wagmi/providers/public'
import { uniposChains } from '../utils/constants/chains'

// Config
const { chains, provider } = configureChains(
  uniposChains,
  [
    infuraProvider({
      apiKey: process.env.INFURA_API_KEY,
    }),
    publicProvider(),
  ],
  {
    targetQuorum: 1,
  }
)

const wagmiClient = createClient({
  autoConnect: true,
  connectors: [
    new InjectedConnector({
      chains,
    }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
  ],
  provider,
})

const WagmiProvider = ({ children }: { children: React.ReactNode }) => {
  return <WagmiConfig client={wagmiClient}>{children}</WagmiConfig>
}

export default WagmiProvider
```
Im Getting this error on Next.js
Is there anny fix for this

**Reply by quankuiyu** [reply]
> 您使用的是什么链条？如果不是`mainnet`或`goerli`，则它们不支持 ENS，您要么需要部署 ENS 合约，要么以另一个链为目标。



> ```
> Unhandled Runtime Error
> Error: unsupported network (argument="network", value={"chainId":137,"name":"polygon"}, code=INVALID_ARGUMENT, version=providers/5.7.2)
> ```
> 
> ```
> // Wagmi
> import { configureChains, createClient, WagmiConfig } from 'wagmi'
> import { InjectedConnector } from 'wagmi/connectors/injected'
> import { WalletConnectConnector } from 'wagmi/connectors/walletConnect'
> import { infuraProvider } from 'wagmi/providers/infura'
> import { publicProvider } from 'wagmi/providers/public'
> import { uniposChains } from '../utils/constants/chains'
> 
> // Config
> const { chains, provider } = configureChains(
>   uniposChains,
>   [
>     infuraProvider({
>       apiKey: process.env.INFURA_API_KEY,
>     }),
>     publicProvider(),
>   ],
>   {
>     targetQuorum: 1,
>   }
> )
> 
> const wagmiClient = createClient({
>   autoConnect: true,
>   connectors: [
>     new InjectedConnector({
>       chains,
>     }),
>     new WalletConnectConnector({
>       chains,
>       options: {
>         qrcode: true,
>       },
>     }),
>   ],
>   provider,
> })
> 
> const WagmiProvider = ({ children }: { children: React.ReactNode }) => {
>   return <WagmiConfig client={wagmiClient}>{children}</WagmiConfig>
> }
> 
> export default WagmiProvider
> ```
> 
> 我在 Next.js 上收到此错误 是否有任何修复

You need to introduce polygon in configureChains
